scripts/trace_docs_deps.py

```diff
@@ -1,116 +1,181 @@
 """
-Usage: `python trace_docs_deps.py [DOCS_DIR]`
+Usage: `python trace_docs_deps.py`
 
 This script is used in our Azure Docs Integration pipeline (azure-pipelines-docs-integration.yml) to determine whether
 a change has been made in the `great_expectations/` directory that change impacts `docs/` and the snippets therein.
 
 The script takes the following steps:
-    1. Parses all markdown files in `docs/`, using regex to find any Docusaurus links (i.e. ```python file=...#L10-20)
-    2. Goes to each linked file and uses AST to parse imports used there
-    3. Filters for only relative imports and determines the paths to those files
+    1. Uses AST to parse the source code in `great_expectations/`; the result is a mapping between function/class definition and the origin file of that symbol
+    2. Parses all markdown files in `docs/`, using regex to find any Docusaurus links (i.e. ```python file=...#L10-20)
+    3. Evaluates each linked file using AST and leverages the definition map from step #1 to determine which source files are relevant to docs under test
 
 The resulting output list is all of the dependencies `docs/` has on the primary `great_expectations/` directory.
 If a change is identified in any of these files during the pipeline runtime, we know that a docs dependency has possibly
 been impacted and the pipeline should run to ensure adequate test coverage.
 
-Please note that this solution has some inherit disclaimers one should be familiar with. As there is no actual execution
-of source code (but rather parsing of the source file's AST), it is hard to pinpoint the location of a given import with
-absolute certainty.  Utilizing the `__file__` attr would provide the exact path of an import but without that, we need
-to assume that an `ast.Import` node corresponds to an accurate path in our codebase (ex: `great_expectations.core.batch`
-results in great_expectations/core/batch.py).
-
-As long as imports correspond to the relative path of modules, this method will continue to work as intended. Deviations
-here could result in certain imports not being picked up by the script, reducing the accuracy and frequency of the
-Azure Docs Integration pipeline's triggers. As GE has maintained this consistency throughout its development, this
-script and the methods used herein are deemed to be appropriate.
-
 """
```

Comment:
The prior method had some ambiguity and required imports to match the file structure.

This new approach captures all function/class definitions in the source code and if we happen to run into one of those definitions in our docs under test, we make an association.

Reply:
Thank you for this additional clarity 🙇🏼